Microsoft.AVS.VMFS/Microsoft.AVS.VMFS.psm1

```diff
@@ -1083,3 +1092,77 @@ function Mount-VmfsDatastore {
     }
                
   }
+
+
+<#
+    .SYNOPSIS
+     This function list all VMFS datastores accessible to host(s) under the given ESXi Cluster.
+
+     1. vSphere Cluster Name
+          
+    .PARAMETER ClusterName
+     vSphere Cluster Name
+    
+    .EXAMPLE
+     Get-VmfsDatastore -ClusterName "vSphere-cluster-001"  
+
+    .INPUTS
+     vSphere Cluster Name
+
+    .OUTPUTS
+     None.
+#>
+
+function Get-VmfsDatastore {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+   
+    Param
+    (
+      [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'vSphere Cluster Name')]
+      [string] $ClusterName
+
+    )
+       
+    Write-Host "Collecting all available VMFS datastores accessible to ESXi host(s) in the cluster "  $ClusterName
+    Write-Host ""    
+    $ClusterName = $ClusterName.Trim()
+
+    $Cluster = Get-Cluster -Name $ClusterName -ErrorAction Ignore
+    if (-not $Cluster) {
+        throw "Cluster $($ClusterName) does not exist."
+    }
+
+    $VmHosts = $Cluster | Get-VMHost -ErrorAction Ignore 
+    if (-not $VmHosts) {
+        throw "No ESXi host found under $($ClusterName)."
+    }
+
+
+    $Datastores = Get-VMHost -Name $VmHosts | Get-Datastore | Where-Object {$_.Type -match "VMFS"} | Get-Unique
+
+    if ( -not $Datastores) {
+        Write-Host "No Datastore found under the given cluster."
+        return
+      
+    }
+
+    foreach ($Datastore in $Datastores){
+
+      $VmfsUuid = $Datastore.ExtensionData.info.Vmfs.uuid 
+      $HostViewDiskName = $Datastore.ExtensionData.Info.vmfs.extent[0].Diskname;
+      
+      Write-Output "Name : $($Datastore.Name)"
```

Comment:
I'd suggest also storing it in [`NamedOutputs`](https://github.com/Azure/Microsoft.AVS.Management/tree/main/docs#outputs-and-termination-of-a-function) so that automation could easily read it.

Reply:
that makes sense. I'll update this.

Reply:
fixed.